lib/internal/readline/interface.js

```diff
@@ -1324,41 +1321,15 @@ class Interface extends InterfaceConstructor {
    * @returns {InterfaceAsyncIterator}
    */
   [SymbolAsyncIterator]() {
-    if (this[kLineObjectStream] === undefined) {
-      if (Readable === undefined) {
-        Readable = require('stream').Readable;
-      }
-      const readable = new Readable({
-        objectMode: true,
-        read: () => {
-          this.resume();
-        },
-        destroy: (err, cb) => {
-          this.off('line', lineListener);
-          this.off('close', closeListener);
-          this.close();
-          cb(err);
-        },
-      });
-      const lineListener = (input) => {
-        if (!readable.push(input)) {
-          // TODO(rexagod): drain to resume flow
-          this.pause();
-        }
-      };
-      const closeListener = () => {
-        readable.push(null);
-      };
-      const errorListener = (err) => {
-        readable.destroy(err);
-      };
-      this.on('error', errorListener);
-      this.on('line', lineListener);
-      this.on('close', closeListener);
-      this[kLineObjectStream] = readable;
+    if (!this[kLineObjectStream]) {
+      this[kLineObjectStream] = EventEmitter.on(
+        this, 'line', {
+          close: ['close'],
+          pauseThreshold: 1024,
```

Comment:
pauseThreshold -> highWaterMark?

Reply:
Yeah, Could it be pauseThreshold -> highWaterMark and resumeThreshold -> lowWaterMark? I understood that I got the best result in my testing controlling the moment to pause and to resume with two different thresholds

Reply:
@ronag what do you think (about naming)?

(to tl;dr; this PR improves the performance of readline's async iterator and `events.on` by making the implementation faster using a FixedQueue and exposing machinery to deal with backpressure in `on` that isn't exposed now but will be in a future PR)

(ideally readline would just be a stream but that probably can't work)